packages/lib-sourcify/src/lib/solidityCompiler.ts

```diff
@@ -29,40 +42,47 @@ export async function useCompiler(version: string, solcJsonInput: JsonInput) {
   // Not possible to retrieve compilers with "-ci.".
   if (version.includes('-ci.')) version = version.replace('-ci.', '-nightly.');
   const inputStringified = JSON.stringify(solcJsonInput);
-  const solcPath = await getSolcExecutable(version);
   let compiled: string | undefined;
 
-  if (solcPath) {
-    const shellOutputBuffer = spawnSync(solcPath, ['--standard-json'], {
-      input: inputStringified,
-      maxBuffer: 1000 * 1000 * 10,
-    });
-
-    // Handle errors.
-    let error: false | Error = false;
-    if (shellOutputBuffer.error) {
-      const typedError: NodeJS.ErrnoException = shellOutputBuffer.error;
-      // Handle compilation output size > stdout buffer
-      if (typedError.code === 'ENOBUFS') {
-        error = new Error('Compilation output size too large');
+  const solcPlatform = findSolcPlatform();
+  if (solcPlatform) {
+    const solcPath = await getSolcExecutable(solcPlatform, version);
+    if (solcPath) {
```

Comment:
If `solcPath` is assigned `false` here, because of the validation, it would silently fail and not fall back. I suggest:

```ts
  const solcPlatform = findSolcPlatform();
  let solcPath;
  if (solcPlatform) {
    solcPath = await getSolcExecutable(solcPlatform, version);
  }
  if (solcPath) {
    ...
  } else {
    const soljson = await getSolcJs(version);
    if (soljson) {
      compiled = soljson.compile(inputStringified);
    }
  }
```